src/execution/vm/bytecode_generator.cpp

```diff
@@ -3484,6 +3496,20 @@ LocalVar BytecodeGenerator::NewStaticString(ast::Context *ctx, const ast::Identi
   return static_local;
 }
 
+LocalVar BytecodeGenerator::CreateArgumentArray(const util::RegionVector<execution::ast::Expr *> &arguments,
+                                                const uint32_t start, const uint32_t end, execution::ast::Type *type) {
+  auto num_args = end - start;
+  const auto array_type = ast::ArrayType::Get(num_args, type->PointerTo());
+  LocalVar args = GetCurrentFunction()->NewLocal(array_type);
+  auto arr_elem_ptr = GetCurrentFunction()->NewLocal(type->PointerTo()->PointerTo());
+  for (uint32_t i = 0; i < num_args; i++) {
+    GetEmitter()->EmitLea(arr_elem_ptr, args, i * 8);
+    LocalVar input_string = VisitExpressionForLValue(arguments[i + start]);
```

Comment:
It seems like you're using this for the right-hand side of an assignment. Use VisitExpressionForRValue instead as it looks like this being treated like an R-value.

Reply:
So for this I'm a little confused tbh. @pmenon can you weigh in here? I thought that these were supposed to be LValues and we were mistakenly using RValues. @pmenon recently opened this issue: https://github.com/cmu-db/noisepage/issues/1167

Reply:
SQL values should use L-values. But, this function shouldn't make assumptions on how to derive child expressions. Either (1) add a boolean indicating whether to use L- or R-values, or (2) pull this logic into `CONCAT` generation. I'd prefer the latter until we have enough use-cases for this to make refactor into a common function.

Reply:
I pulled it into CONCAT generation, and kept it as L-Value (which I think is what you were saying).